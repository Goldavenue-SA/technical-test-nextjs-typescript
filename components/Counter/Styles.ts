import styled from "styled-components";

export const StyledSection = styled.section`
    display: flex;
    flex-direction: row;
    gap: var(--global-rhythm);

  p:last-of-type {
    margin-bottom: 0;
  }
`;
