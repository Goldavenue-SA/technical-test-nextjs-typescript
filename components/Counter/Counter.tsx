import { StyledSection } from './Styles';

interface CounterProps {
  countOverThreshold: number;
  min: number;
  max: number;
}

export const Counter = ({
  countOverThreshold = 0,
  min = 0,
  max = 0,
}: CounterProps) => {
  return (
    <>
      <StyledSection>
        <p role="status" aria-live="polite">
          Count over threshold:{' '}
          <span>
            {' '}
            <strong>{countOverThreshold}</strong>{' '}
          </span>
        </p>
        <p>
          Min:{' '}
          <span>
            {' '}
            <strong>{min}</strong>{' '}
          </span>
        </p>
        <p>
          Max:{' '}
          <span>
            {' '}
            <strong>{max}</strong>{' '}
          </span>
        </p>
      </StyledSection>
    </>
  );
};
