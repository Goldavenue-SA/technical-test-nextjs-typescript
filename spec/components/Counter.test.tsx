import { render, screen } from '@testing-library/react';
import { Counter, CounterProps } from '../../components/Counter/Counter';
import '../setupTests';

describe('Test Counter Component', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  const counterProps: CounterProps = {
    countOverThreshold: 239,
    max: 235,
    min: 100,
  };

  const setupComponent = () => render(<Counter {...counterProps} />);

  it('renders correctly and matches the snapshot', () => {
    const { asFragment } = setupComponent();

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should render the component with the correct props', () => {
    setupComponent();

    expect(
      screen
        .getByText('Count over threshold:')
        .closest('p')
        ?.querySelector('span > strong')
    ).toHaveTextContent(counterProps.countOverThreshold.toString());

    expect(
      screen.getByText('Min:').closest('p')?.querySelector('span > strong')
    ).toHaveTextContent(counterProps.min.toString());

    expect(
      screen.getByText('Max:').closest('p')?.querySelector('span > strong')
    ).toHaveTextContent(counterProps.max.toString());
  });
});
