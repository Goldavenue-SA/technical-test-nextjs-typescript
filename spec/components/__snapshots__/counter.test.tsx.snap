// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Counter Component renders correctly and matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="sc-5d04bb7b-0 iLyYNB"
  >
    <p
      aria-live="polite"
      role="status"
    >
      Count over threshold:
      <span>
        <strong>
          239
        </strong>
      </span>
    </p>
    <p>
      Min:
      <span>
        <strong>
          100
        </strong>
      </span>
    </p>
    <p>
      Max:
      <span>
        <strong>
          235
        </strong>
      </span>
    </p>
  </section>
</DocumentFragment>
`;
